src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/WhyCommand/WhyCommandUtility/DependencyGraphFinder.cs

```diff
@@ -0,0 +1,306 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using NuGet.ProjectModel;
+
+namespace NuGet.CommandLine.XPlat.WhyCommandUtility
+{
+    internal static class DependencyGraphFinder
+    {
+        /// <summary>
+        /// Finds all dependency graphs for a given project.
+        /// </summary>
+        /// <param name="assetsFile">Assets file for the project.</param>
+        /// <param name="targetPackage">The package we want the dependency paths for.</param>
+        /// <param name="userInputFrameworks">List of target framework aliases.</param>
+        /// <returns>
+        /// Dictionary mapping target framework aliases to their respective dependency graphs.
+        /// Returns null if the project does not have a dependency on the target package.
+        /// </returns>
+        public static Dictionary<string, List<DependencyNode>?>? GetAllDependencyGraphs(
+            LockFile assetsFile,
+            string targetPackage,
+            List<string> userInputFrameworks)
+        {
+            var dependencyGraphPerFramework = new Dictionary<string, List<DependencyNode>?>(assetsFile.Targets.Count);
+            bool doesProjectHaveDependencyOnPackage = false;
+
+            // get all top-level package and project references for the project, categorized by target framework alias
+            Dictionary<string, List<string>> topLevelReferencesByFramework = GetTopLevelPackageAndProjectReferences(assetsFile, userInputFrameworks);
+
+            if (topLevelReferencesByFramework.Count > 0)
+            {
+                foreach (var (targetFrameworkAlias, topLevelReferences) in topLevelReferencesByFramework)
+                {
+                    LockFileTarget target = assetsFile.GetTarget(targetFrameworkAlias, runtimeIdentifier: null);
+
+                    if (target != null)
```

Comment:
Is the idea that we would've validated that this target exists in the inputs? 

I think we might just want to assume it exists here and let is NRE in case made a mistake earlier. 

This potentially hides it.

Reply:
We're now validating that the input frameworks have a corresponding target at an earlier point, so I've removed this check from here. I've also added a test for the warning output from the validation